app/src/main/java/com/lastaoutdoor/lasta/data/db/SocialDBRepositoryImpl.kt

```diff
@@ -25,14 +29,42 @@ class SocialDBRepositoryImpl @Inject constructor(context: Context, database: Fir
 
   override suspend fun getFriends(friendIds: List<String>): List<UserModel> {
     val friends: ArrayList<UserModel> = ArrayList()
-    if (friendIds.size == 1) return friends
     val friendsQuery = userCollection.whereIn(FieldPath.documentId(), friendIds).get().await()

```

Comment:
You still have to check that the friendsId list is not empty, otherwise the whereIn query will fail and not be handled correctly.

Reply:
I did so and added the condition `friendsID[0] == ""`, else it would work for empty list but not if you have one friend